### PR TITLE
Cleanup

### PR DESCRIPTION
**Pull request by wardog1uk**
I did a bunch of cleaning and refactoring in AwfulApplication.java, plus a little in NetworkUtils.java and ThemeSettings.java.

I've moved all the font stuff from AwfulApplication to a new FontManager class and passed through the calls. It  probably would be better to have it somewhere else, or maybe as a singleton.

There is definitely an existing bug with the action bar's font occasionally not being updated when it gets changed, but I can't reliably reproduce it.

### CODE REVIEW

**Comment by baka-kaba** [comment]
Oh hey thanks! I had no idea this PR existed until now

I'll check over it as soon as I get time, looks good from scanning over it though

File: Awful.apk/src/main/java/com/ferg/awfulapp/FontManager.java [hunk 1]
```diff
@@ -30,7 +30,29 @@
      * @param assets        An AssetManager for accessing the font files
      */
     public FontManager(String preferredFont, AssetManager assets) {
-        buildFontList(preferredFont, assets);
```
[the remark this answers is not in the document]
Comment (baka-kaba) [reply Awful.apk/src/main/java/com/ferg/awfulapp/FontManager.java]: I think I messed up with the comment position on this one but you get what I mean!

**Comment by baka-kaba** [comment]
Also how do you want to be credited in the contributors thing?

**Comment by wardog1uk** [comment]
Oben. I just pushed some more changes I did earlier and will sort out the new bits in a minute.

**Comment by wardog1uk** [comment]
Ok I think I'm done.

**Pull request merged**